src/main/scala/com/vngrs/etl/Transform.scala

```diff
@@ -0,0 +1,64 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * Transforms data ([[org.apache.spark.rdd.RDD]]) to another form.
+  *
+  * @tparam A Type of the data ([[org.apache.spark.rdd.RDD]]).
+  */
+trait Transform[A, B] {
+
+  /**
+    * Runs transformation operation,
+    *
+    * @param input [[org.apache.spark.rdd.RDD]]
+    * @return Transformed [[org.apache.spark.rdd.RDD]]
+    */
+  def apply(input: RDD[A]): RDD[B]
+}
+
+/**
+  * Companion object which acts as a Factory.
+  */
+object Transform {
```

Comment:
Yes, exception is totally normal. What I meant was we can not keep mapper method name `apply` if we going to implement a generic factory method.

A generic is required since only transformation is not just a mapping but a filtering as well. If you think like this then a generic one is required. Also, giving mapper function a name - `map` - instead of  `apply` is more descriptive like `filter`.